### PR TITLE
Feat: custom substitute types

### PR DESCRIPTION
**Pull request by Will-Mann-16**
Added ability to add custom substitute types in the optional configuration parameters. This is to be used with a custom MarshalJSON functions setup. It will replace the top level component type sbut leave all the other refs intact.

Also removed passthrough logic to allow for openapi spec to be in complete control, and means we don't necessarily have to account for all these types ourselves (think sql.NullString, errors, time.time or anything else).

### CODE REVIEW

File: outputs/openapi/components.go [hunk 1]
```diff
@@ -182,7 +186,7 @@ func componentToSchema(component astra.Field, bindingType astTraversal.BindingTa
 			}
 		}
 	} else if component.Type == "slice" {
-		itemSchema := mapAcceptedType(component.SliceType)
+		itemSchema := mapPredefinedTypeFormat(component.Type)
```
[the remark this answers is not in the document]
Comment (Will-Mann-16) [reply outputs/openapi/components.go:189]: Yep that's it good catch

File: outputs/openapi/components.go [hunk 2]
```diff
@@ -198,7 +202,7 @@ func componentToSchema(component astra.Field, bindingType astTraversal.BindingTa
 			Items: &itemSchema,
 		}
 	} else if component.Type == "map" {
-		additionalProperties := mapAcceptedType(component.MapValueType)
+		additionalProperties := mapPredefinedTypeFormat(component.Type)
```
[the remark this answers is not in the document]
Comment (Will-Mann-16) [reply outputs/openapi/components.go:205]: Yes it should

File: outputs/openapi/components.go [hunk 3]
```diff
@@ -212,7 +216,7 @@ func componentToSchema(component astra.Field, bindingType astTraversal.BindingTa
 			AdditionalProperties: &additionalProperties,
 		}
 	} else {
```
[the remark this answers is not in the document]
Comment (Will-Mann-16) [reply outputs/openapi/components.go:218]: No I don't think we do - let me also add an example to the basic example (we really need some integration tests)

[the remark this answers is not in the document]
Comment (Will-Mann-16) [reply outputs/openapi/components.go:235]: Added example in this instance, as it was covered by this new work

File: examples/basic/openapi.generated.yaml [hunk 4]
```diff
@@ -153,7 +154,7 @@ components:
                 comments:
                     type: array
                     items:
-                        $ref: '#/components/schemas/types.Comment'
+                        type: array
```
[the remark this answers is not in the document]
Comment (Will-Mann-16) [reply examples/basic/openapi.generated.yaml]: Fixed now

**Review by TommoLeedsy — APPROVED** [review]

**Pull request merged**